### PR TITLE
[iam] Setup chi mux for routing

### PR DESCRIPTION
**Pull request by easyCZ**
## Related Issue(s)

Fixes #

## How to test

Run locally with `cd components/iam && go run main.go run` and visit localhost:9002/oidc/start and observe hello printed

## Release Notes

```release-note
NONE
```

## Werft options:

- [ ] /werft with-local-preview
      If enabled this will build `install/preview`
- [ ] /werft with-preview
- [ ] /werft with-large-vm
- [ ] /werft with-integration-tests=all
      Valid options are `all`, `workspace`, `webapp`, `ide`, `jetbrains`, `vscode`, `ssh`

### CODE REVIEW

File: components/iam/pkg/server/server.go [hunk 1]
```diff
@@ -24,9 +28,30 @@ func Start(logger *logrus.Entry, version string, cfg *config.ServiceConfig) erro
...
 	}
 
 	return nil
 }
+
+func register(srv *baseserver.Server) error {
+	router := mux.NewRouter()
+	mount(router, "/oidc", oidc.Router())
+
+	srv.HTTPMux().Handle("/", router)
```
Comment (AlexTugarev) [inline components/iam/pkg/server/server.go:47]: Is there a reason to not extend `baseserver.Server` for that?

**Comment by AlexTugarev** [comment]
Thanks for pointer with integration of middleware. 

I'd prefer https://github.com/go-chi/chi as a router/middleware tech for two reasons:
1. that seems to be lighter, maybe just because it doesn't have that much history in comparison.
2. https://github.com/gorilla/mux is no longer maintained.

File: components/iam/pkg/server/server.go (hunk 1 above)
Comment (easyCZ) [reply components/iam/pkg/server/server.go:47]: `baseserver.Server` as such doesn't need Chi or gorilla for that matter. In go, all http handlers are ultimately an `http.Handler` and for that reason, we make our choices on the application level and still remain compatible with `http.Handler`. That also means that we shouldn't push this into baseserver as we'd be dictating the use of a "helper" library when the more general interface sufficies.

**Comment by easyCZ** [comment]
Yeah chi is okay. I've not used it here because we've been using gorilla in other parts already.

> [gorilla/mux](https://github.com/gorilla/mux) is no longer maintained.
As of 3 days ago. Will change it to chi but this is news to me also :)

**Review by AlexTugarev — APPROVED** [review]
Awesome! Thanks!

**Pull request merged**